src/ddmd/expression.d

```diff
@@ -15409,7 +15410,21 @@ extern (C++) final class CmpExp : BinExp
             }
             if ((t1.ty == Tarray || t1.ty == Tsarray) && (t2.ty == Tarray || t2.ty == Tsarray))
             {
-                semanticTypeInfo(sc, t1.nextOf());
+                // Lower to object.__cmp(e1, e2)
+                Expression al = new IdentifierExp(loc, Id.empty);
+                al = new DotIdExp(loc, al, Id.object);
+                al = new DotIdExp(loc, al, Id.__cmp);
+                al = al.semantic(sc);
+
+                auto arguments = new Expressions();
+                arguments.push(e1);
+                arguments.push(e2);
+
+                al = new CallExp(loc, al, arguments);
+                al = new CmpExp(op, loc, al, new IntegerExp(0));
+                al = al.semantic(sc);
```

Comment:
If I return ```al``` at the end of this code block, I get the following compilation error output for that test:
```
fail_comp.d(4): Error: floating point operator '!<>=' always returns false for non-floating comparisons
fail_comp.d(5): Error: use '==' for non-floating comparisons rather than floating point operator '!<>'
fail_comp.d(6): Error: use '!=' for non-floating comparisons rather than floating point operator '<>'
fail_comp.d(7): Error: floating point operator '<>=' always returns true for non-floating comparisons
fail_comp.d(8): Error: use '<=' for non-floating comparisons rather than floating point operator '!>'
fail_comp.d(9): Error: use '<' for non-floating comparisons rather than floating point operator '!>='
fail_comp.d(10): Error: use '>=' for non-floating comparisons rather than floating point operator '!<'
fail_comp.d(11): Error: use '>' for non-floating comparisons rather than floating point operator '!<='
fail_comp.d(12): Error: floating point operator '<>=' always returns true for non-floating comparisons
fail_comp.d(13): Error: floating point operator '!<>=' always returns false for non-floating comparisons
```

There is no more duplication, but the output is not precisely the same as the one expected, which is
```
fail_compilation/ice11755.d(20): Error: '!<>=' is not defined for array comparisons
fail_compilation/ice11755.d(21): Error: use '==' for non-floating comparisons rather than floating point operator '!<>'
fail_compilation/ice11755.d(22): Error: use '!=' for non-floating comparisons rather than floating point operator '<>'
fail_compilation/ice11755.d(23): Error: '<>=' is not defined for array comparisons
fail_compilation/ice11755.d(24): Error: use '<=' for non-floating comparisons rather than floating point operator '!>'
fail_compilation/ice11755.d(25): Error: use '<' for non-floating comparisons rather than floating point operator '!>='
fail_compilation/ice11755.d(26): Error: use '>=' for non-floating comparisons rather than floating point operator '!<'
fail_compilation/ice11755.d(27): Error: use '>' for non-floating comparisons rather than floating point operator '!<='
fail_compilation/ice11755.d(28): Error: floating point operator '<>=' always returns true for non-floating comparisons
fail_compilation/ice11755.d(29): Error: floating point operator '!<>=' always returns false for non-floating comparisons
```